ang/civicase/case/actions/services/email-case-action.service.js

```diff
@@ -0,0 +1,156 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase');
+
+  module.service('EmailCaseAction', EmailCaseAction);
+
+  /**
+   * EmailCaseAction service.
+   *
+   * @param {object} $q $q service
+   * @param {object} ts translation service
+   * @param {Function} isTruthy service to check if value is truthy
+   * @param {object} dialogService dialog service
+   * @param {Function} parseUrlParameters parse url parameters
+   * @param {object} $window browsers window object
+   * @param {object} CaseType case type service
+   * @param {object} CaseTypeCategory case type category service
+   * @param {object} civicaseCrmApi service to use civicrm api
+   * @param {Function} getSelect2Value service to get select 2 values
+   */
+  function EmailCaseAction ($q, ts, isTruthy, dialogService, parseUrlParameters,
+    $window, CaseType, CaseTypeCategory, civicaseCrmApi, getSelect2Value) {
+    var model = {};
```

Comment:
Why define `model` globally when you can define it in `doAction` and pass it directly to `openRoleSelectorPopUp` ?